modules/helfi_paragraphs_hearings/src/Plugin/ExternalEntities/StorageClient/Hearings.php

```diff
@@ -0,0 +1,219 @@
+
+  /**
+   * Api base url.
+   *
+   * @var string
+   */
+  public static string $apiUrl = 'https://api.hel.fi/kerrokantasi/v1/hearing?';
+
+  /**
+   * Hearing base url.
+   *
+   * @var string
+   */
+  public static string $hearingUrl = 'https://kerrokantasi.hel.fi/';
+
+  /**
+   * The current language service.
+   *
+   * @var \Drupal\Core\Language\LanguageManagerInterface
+   */
+  private LanguageManagerInterface $languageManager;
+
+  /**
+   * The HTTP client service.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  private ClientInterface $client;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(
+    ContainerInterface $container,
+    array $configuration,
+    $plugin_id,
+    $plugin_definition
+  ) : self {
+    $instance = parent::create($container, $configuration, $plugin_id, $plugin_definition);
+    $instance->languageManager = $container->get('language_manager');
+    $instance->client = $container->get('http_client');
+
+    return $instance;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function loadMultiple(array $ids = NULL) : array {
+    $ids = $ids ?: [];
+
+    $data = $this->query(['ids' => $ids]);
+
+    return $data;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function save(ExternalEntityInterface $entity) : void {
+    throw new EntityStorageException('::save() is not supported.');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function delete(ExternalEntityInterface $entity) : void {
+    throw new EntityStorageException('::delete() is not supported.');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function query(
+    array $parameters = [],
+    array $sorts = [],
+          $start = NULL,
+          $length = NULL
+  ) : array {
+
+    $langcode = $this->languageManager
+      ->getCurrentLanguage(LanguageInterface::TYPE_CONTENT)
+      ->getId();
+
+    $query = http_build_query([
+      'format' => 'json',
+      'langcode' => $langcode,
+      'open' => 'true',
+    ]);
+
+    $uri = sprintf('%s%s', self::$apiUrl, $query);
+
+    try {
+      $content = $this->client->request('GET', $uri);
+      $json = Utils::jsonDecode($content->getBody()->getContents(), TRUE);
+      if (empty($json['results'])) {
+        return [];
+      }
+    }
+    catch (RequestException | GuzzleException $e) {
+      watchdog_exception('helfi_paragraphs_hearings', $e);
+    }
+
+    $count = $json['count'];
+    $results = $count > 3 ? array_slice($json['results'], 0, 3) : $json['results'];
+
+    if (isset($parameters['ids']) && $parameters['ids']) {
+      $items = array_filter($results, function ($item) use ($parameters) {
+        return in_array($item['id'], $parameters['ids']);
+      });
+      $results = $items;
+    }
+
+    $data = [];
+
+    foreach ($results as $hearing) {
+      $existingTranslations = $this->getTranslationLanguages($hearing);
+      $selectedLangcode = $this->resolveLanguage($hearing, $langcode);
+
+      $item = [
```

Comment:
This can be simplified to 
```php
$item = $hearing;
$item += [
        'main_image_url' => $hearing['main_image']['url'],
        'main_image' => Url::fromUri($hearing['main_image']['url']),
        'count' => $count,
        'url' => sprintf('%s%s', self::$hearingUrl, $hearing['slug']),
        'langcode' => $selectedLangcode,
        'existing_translations' => implode(',', $existingTranslations),
];
```